pkg/contracts/src/OsmoticController.sol

```diff
@@ -107,146 +79,63 @@ contract OsmoticController is Initializable, OwnableUpgradeable, PausableUpgrade
-    /* ** Locking Functions                                                                                                              ***/
-    /* *************************************************************************************************************************************/
-
-    function lockBalance(address _token, uint256 _amount) public whenNotPaused {
-        IStaking staking = stakingFactory.getOrCreateInstance(_token);
-
-        _lockBalance(staking, msg.sender, _amount);
-    }
-
-    function unlockBalance(address _token) public whenNotPaused {
-        require(participantSupportedPools[msg.sender] == 0, "OsmoticController: cannot unlock while supporting a pool");
-
-        (uint256 lockedBalance,) = getStakingLock(_token, msg.sender);
-
-        _unlockBalance(stakingFactory.getInstance(_token), msg.sender, lockedBalance);
-    }
-
-    /* *************************************************************************************************************************************/
-    /* ** Locking and Supporting Functions                                                                                               ***/
-    /* *************************************************************************************************************************************/
-
-    function lockAndSupport(
-        OsmoticPool _pool,
-        uint256 _lockedAmount,
-        ParticipantSupportUpdate[] calldata _participantUpdates
-    ) external whenNotPaused {
-        lockBalance(address(_pool.governanceToken()), _lockedAmount);
-
-        _pool.updateProjectSupports(_participantUpdates);
-    }
-
-    function unsupportAndUnlock(OsmoticPool _pool, ParticipantSupportUpdate[] calldata _participantUpdates)
+    function createMimeToken(string calldata name, string calldata symbol, bytes32 merkleRoot)
         external
         whenNotPaused
+        returns (address)
     {
-        _pool.updateProjectSupports(_participantUpdates);
-
-        unlockBalance(address(_pool.governanceToken()));
-    }
-
-    /* *************************************************************************************************************************************/
-    /* ** Participant Supported Pools Functions                                                                                          ***/
-    /* *************************************************************************************************************************************/
-
-    function increaseParticipantSupportedPools(address _participant) external whenNotPaused onlyPool(msg.sender) {
-        participantSupportedPools[_participant]++;
-
-        emit ParticipantSupportedPoolsChanged(_participant, participantSupportedPools[_participant]);
-    }
+        MimeToken token = MimeTokenFactory(mimeTokenFactory).createMimeToken(name, symbol, merkleRoot);
+        token.transferOwnership(msg.sender);
+
+        return address(token);
+    }
+
+    function createPool(
```

Comment:
It would be interesting to have a way to create a pool with an already-existing token.

Maybe we could have a `createPoolWithMimeToken` function of some sort.

wdyt?

Reply:
Yes, I can include the extra method 👍